src/commands/dev/lib/types.ts

```diff
@@ -109,6 +111,9 @@ export interface Builder {
   prepareCache?(
     params: PrepareCacheParams
   ): CacheOutputs | Promise<CacheOutputs>;
+  [PACKAGE]?: {
+    version: string;
+  };
```

Comment:
Instead of modifying this interface, perhaps a better type-safe way is to create a new interface like

```ts
export interface BuilderWithVersion extends Builder {
[PACKAGE]: {
    version: string;
  }
}
```

This will do two things:

1. Allow us to eventually move the `Builder` interface into `build-utils` where it belongs.
2. You shouldn't have to use the bang `!` syntax to get TS to understand the package exists.

Reply:
Another alternative which is arguably safer is to use the following:

```ts
export interface BuilderWithVersion {
    builder: Builder;
    version: string;
}
```

Reply:
Good call, I'll refactor to use this API but in a different PR.